File: lab-service-material/src/main/java/com/lab/material/entity/Material.java
package com.lab.material.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 药品信息实体
 */
@Data
@TableName("material")
public class Material {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private String materialCode;
    
    private String materialName;
    
    private Integer materialType;
    
    private Long categoryId;
    
    private String specification;
    
    private String unit;
    
    private String casNumber;
    
    private String dangerCategory;
    
    private Integer isControlled;
    
    private Long supplierId;
    
    private BigDecimal unitPrice;
    
    private Integer safetyStock;
    
    private Integer maxStock;
    
    private String storageCondition;
    
    private Integer shelfLifeDays;
    
    private String description;
    
    private String imageUrl;
    
    private Integer status;
    
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedTime;
    
    @TableLogic
    private Integer deleted;
}

File: lab-web-admin/src/types/material.ts
export interface Material {
  id: number
  materialCode: string
  materialName: string
  materialType: number // 1-耗材 2-试剂 3-危化品
  categoryId: number
  categoryName?: string
  specification?: string
  unit: string
  manufacturer?: string
  supplierId?: number
  supplierName?: string
  unitPrice?: number
  safetyStock?: number
  imageUrl?: string
  casNumber?: string
  dangerCategory?: string
  isControlled?: number // 0-非管控 1-易制毒 2-易制爆
  storageConditions?: string
  safetyInstructions?: string
  status: number
  createdTime?: string
  updatedTime?: string
}

export interface MaterialQuery {
  keyword?: string
  materialType?: number
  categoryId?: number
  isControlled?: number
  status?: number
  page?: number
  size?: number
}

export interface MaterialForm {
  id?: number
  materialCode: string
  materialName: string
  materialType: number
  categoryId: number
  specification?: string
  unit: string
  manufacturer?: string
  supplierId?: number
  unitPrice?: number
  imageUrl?: string
  casNumber?: string
  dangerCategory?: string
  isControlled?: number
  storageConditions?: string
  safetyInstructions?: string
  status?: number
}

export interface MaterialCategory {
  id: number
  categoryName: string
  parentId?: number
  children?: MaterialCategory[]
}

export interface Supplier {
  id: number
  supplierName: string
  contactPerson?: string
  contactPhone?: string
  contactEmail?: string
  address?: string
  status: number
  createdTime?: string
}
